o2.js

```diff
@@ -4,18 +4,56 @@ var shell = require('shell');
 var ipc = require('ipc');
 var fs = require('fs-promise');
 var app = require('app');
+var _ = require('underscore');
+var sanitize = require('sanitize-filename');
+var nodePath = require('path');
+var foldToAscii = require('fold-to-ascii');
+
 var mbOptions = {"width": 400, "height": 400};
 
+
 var mb = menuBar(mbOptions);
 var webContents = null;
 
+var baseUrl = 'https://staging-api.osf.io/v2/';
+
 
-var showNodes = function(nodes){
+var showNodes = function(nodes) {
     "use strict";
     console.log('sending getNodes to ui');
     mb.window.send('getNodes', nodes);
 };
 
+var getNodeFiles = function(nodeId) {
+  console.log('Getting node files');
+  var files = {};
+  mb.window._client.methods.nodeFiles({'path':{'id': nodeId}}, function(data, response) {
+    var fileData = JSON.parse(data.toString());
+    var increment = 1;
+    _.each(fileData.data, function(file) {
+      var safeFilename;
+      var sanitizedName = sanitize(foldToAscii.fold(file.attributes.name));
+      var currentFilenames = _.keys(files);
+      if(_.contains(currentFilenames, sanitizedName)) {
+        var parsedName = nodePath.parse(sanitizedName);
+        safeFilename = parsedName.name+'_'+increment+parsedName.ext;
+        increment+=1;
+      } else {
+        safeFilename = sanitizedName;
+      }
+      // if (safeFilename != file.attributes.name) {
+      //   mb._window.
+      // }
+      files[safeFilename] = _.extend(file.attributes, file.links);
+    });
+  });
+};
+
+ipc.on('did-select-node', function(ev, nodeId) {
```

Comment:
This is it. You're overwriting a did-select-node later on. Merge those together.